### PR TITLE
22956 - Verify update legal filings job to use new db versioning

### PR DESCRIPTION
**Pull request by eason-pan-bc**
*Issue #:* /bcgov/entity#22956

*Description of changes:*
update-legal-filings-job doesn't directly operate in lear db, instead, it's calling legal-api and colin-api endpoints. And there isn't any existing code initialize a db instance in this job. As per discussion I have removed FF for update-legal-filings in multiple flags.json files.

### Verification 1 - Running local legal-api and check behaviours
Keep the local lear-db's colin_last_update table remain the same for 3 runs in different scenarios.
In summary, there is no behaviour differences between these 3 scenarios:
1. Not using new db versioning
Not using new db versioning
2. Using new db versioning and FF on
FF-on
3. Using new db versioning and FF off
FF-off

### Verification 2 - Running local legal-api and verify the endpoints the job used
1. get max colin event_id from lear-db `{{url}}/api/v2/businesses/internal/filings/colin_id`
Pasted image 20241203095542
2. Pick an event_ids greater than the max id returned -- `105911836`
3. Get business info in lear-db: `{{url}}/api/v2/businesses/:identifier`
Pasted image 20241203101654
4. Check whether that event has been updated to lear-db or not: `{{url}}/api/v2/businesses/internal/filings/colin_id/:event_id`
As this event/filing is not yet updated, it's not found in lear-db
Pasted image 20241203103005
5. Update it in lear: `{{url}}/api/v2/businesses/:identifier/filings`
After fixing the data-type for `goodStanding` and `adminFreeze`
Pasted image 20241203104455
6. Update the new max_event_id: `{{url}}/api/v2/businesses/internal/filings/colin_id/:new_max_event_id`
Pasted image 20241203104635
Pasted image 20241203104656
7. Check again (prev check in step-4): `{{url}}/api/v2/businesses/internal/filings/colin_id/:event_id`
And since it's updated to lear-db now, we can find it
image

### Other findings
Found this job is already not really working properly, and might be like this for a while. I left some details here: https://github.com/bcgov/entity/issues/22956#issuecomment-2508692615
May need to cut another ticket to address this issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/update-legal-filings/requirements.txt [hunk 1]
```diff
@@ -28,6 +28,11 @@ asyncio-nats-streaming==0.4.0
...
-git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
+
+# install updated legal-api from feature branch for development, 
+# TODO - change it back when the db versioning feature branch merged to main
+git+https://github.com/bcgov/lear.git@feature-db-versioning#egg=legal_api&subdirectory=legal-api
+# git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
+
 git+https://github.com/bcgov/lear.git#egg=sql-versioning&subdirectory=python/common/sql-versioning
-git+https://github.com/bcgov/business-schemas.git@2.5.12#egg=registry_schemas
+git+https://github.com/bcgov/business-schemas.git@2.18.27#egg=registry_schemas
```
Comment (eason-pan-bc) [inline jobs/update-legal-filings/requirements.txt:38]: update the schema version to the latest (same as legal-api)

**Comment by eason-pan-bc** [comment]
As there is no unit tests for update-legal-filings-job, it shows as failed.
image

**Review by AimeeGao — APPROVED** [review]
LGTM(^-^)

**Comment by eason-pan-bc** [comment]
Did some debugging, and found 2 types of errors:
**1 - caused by incompatible data-type**
```

"errors": [
    {
      "context": [], 
      "error": "None is not of type 'boolean'", 
      "path": "filing/business/goodStanding"
    }, 
    {
      "context": [], 
      "error": "'False' is not of type 'boolean'", 
      "path": "filing/business/adminFreeze"
    }
  ]
```
2 - caused by the business has a draft, or is frozen, or is dissolved in LEAR

**Review by leodube-aot — APPROVED** [review]

**Pull request merged**